server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -929,20 +930,20 @@ public Builder settings(Settings settings) {
             return this;
         }
 
-        public MappingMetaData mapping(String type) {
-            return mappings.get(type);
+        public MappingMetaData mapping() {
+            return mappings.get(MapperService.SINGLE_MAPPING_NAME);
         }
 
-        // TODO remove type here
-        public Builder putMapping(String type, String source) throws IOException {
-            putMapping(new MappingMetaData(type, XContentHelper.convertToMap(XContentFactory.xContent(source), source, true)));
+        public Builder putMapping(String source) {
+            putMapping(new MappingMetaData(MapperService.SINGLE_MAPPING_NAME,
```

Comment:
Is the plan eventually to get rid of the map `ImmutableOpenMap.Builder<String, MappingMetaData> mappings` to instead use just  a single `MappingMetaData mapping`?

Reply:
That's the plan, but it will not be entirely straightforward due to having to deal with sending Diffs of Maps for backwards compatibility, so I've deferred it to a later PR.